lte/gateway/c/core/oai/test/amf/test_amf_procedures.cpp

```diff
@@ -437,6 +436,10 @@ TEST_F(AMFAppProcedureTest, TestRegistrationProcNoTMSI) {
                                                sizeof(ue_smc_response_hexbuf));
   EXPECT_TRUE(rc == RETURNok);
 
+  s6a_update_location_ans_t ula_ans = util_amf_send_s6a_ula(imsi);
```

Comment:
Can we add more scenario, like currently it's assumed that HSS always sends DIAMETER_SUCCESS. 
Scenario-1: If result code other than DIAMETER_SUCCESS
Scenario-2: If HSS didn't provide apn
Scenario-2: If imsi doesn't match

Reply:
this test case only for checking the TestRegistrationProcNoTMSI use case. For the negative scenarios which you have mentioned are already taken care in other UT cases.

You can find them amf_app_test_util.cpp:send_proc_authentication_info_answer().

Reply:
As part of amf_app_test_util.cpp:send_proc_authentication_info_answer(), daimeter cause is taken care for s6a-Authentication information procedure. My concern is, can we add similar case for Update location procedure. Because resule code is not taken care at, https://github.com/magma/magma/blob/master/lte/gateway/c/core/oai/tasks/amf/nas_proc.cpp#L903
On reception of Update Location Ans, amf shall check for result code, if it's success. I think amf should reject the Registration request